sklearn/preprocessing/_polynomial.py

```diff
@@ -979,8 +1022,10 @@ def transform(self, X):
         else:
             XBS = np.zeros((n_samples, n_out), dtype=dtype, order=self.order)
 
-        for i in range(n_features):
-            spl = self.bsplines_[i]
+        for feature_idx in range(n_features):
```

Comment:
As info (non-actionable, too late for that): This renaming of the index from `i` to `feature_idx` creates a lot of noise and line breaks, rendering this PR quite hard to review.